frontend/src/components/SepForm.js

```diff
@@ -16,19 +27,329 @@ const useStyles = makeStyles({
     padding: "2px 0",
     color: "#086375",
   },
-})
+  errorMessage: {
+    marginTop: theme.spacing(1),
+    color: "red",
+  },
+  siteId: {
+    marginTop: "0",
+    marginBottom: theme.spacing(4),
+  },
+  needlesIn: {
+    paddingRight: theme.spacing(1),
+  },
+  needlesOut: {
+    paddingLeft: theme.spacing(1),
+  },
+  clearButton: {
+    marginLeft: theme.spacing(1),
+  },
+}))
 
-const SepForm = () => {
+const SepForm = observer(() => {
   const classes = useStyles()
+  const rootStore = useContext(rootStoreContext)
+  const participantStore = rootStore.ParticipantStore
+  const [sites, setSites] = useState([])
+  const [participantId, setParticipantId] = useState()
+  const SEPFormRef = useRef()
+
+  useEffect(() => {
+    async function fetchSites() {
+      await participantStore.getSites()
+      setSites(participantStore.sites)
+    }
+    fetchSites()
+  }, [])
+
+  const handleClear = () => {
+    setParticipantId(null)
+    if (SEPFormRef.current) {
+      SEPFormRef.current.resetForm()
+    }
+  }
+
   return (
     <Container className={classes.root}>
-      <Grid container component="form" className={classes.form}>
-        <PrevPointHeading className={classes.heading}>
-          SEP Search
-        </PrevPointHeading>
-      </Grid>
+      <Formik
+        className={classes.form}
+        validateOnChange={false}
+        validateOnBlur={false}
+        innerRef={SEPFormRef}
+        initialValues={{
+          site_id: participantStore.currentSite,
+          sep_id: "",
+          last_name: "",
+          date_of_birth: "",
+          maiden_name: "",
+        }}
+        validationSchema={SEPSearchSchema}
+        onSubmit={async (values, { setSubmitting, setFieldValue }) => {
+          await participantStore.getParticipants({
+            sep_id: values.sep_id,
+            last_name: values.last_name,
+            dob: values.date_of_birth,
+            maiden_name: values.maiden_name,
+          })
+          setSubmitting(false)
+          if (participantStore.participants.length === 1) {
+            setParticipantId(participantStore.participants[0].pp_id)
+            setFieldValue("sep_id", participantStore.participants[0].sep_id)
```

Comment:
The reason you get the participant should id, got str error is you want to `setParticipantId()` to `participantStore.participants[0].id`, not `pp_id`. pp id is preventionpoint's internal id, while just `id` is the database id

Reply:
Thank you. I have updated this.